src/document/position.js

```diff
@@ -5,7 +5,12 @@
 
 'use strict';
 
-CKEDITOR.define( [ 'utils', 'ckeditorerror' ], ( utils, CKEditorError ) => {
+CKEDITOR.define( [ 'document/rootelement', 'utils', 'ckeditorerror' ], ( RootElement, utils, CKEditorError ) => {
+	const SAME = 0;
+	const AFTER = 1;
+	const BEFORE = -1;
+	const DIFFERENT = -2;
```

Comment:
OK, @scofalik says that `-1, 0, 1` can be passed to `Array#sort()`. It sounds useful, but it's a bit hacky. Especially that `DIFFERENT` should most likely result in an error being thrown, as this means that positions come from different documents (or something). Therefore, I would propose to normalise these so `compareWith()` returns `0, 1, 2, 3` and we could add a static method to `Position` which could be used to generate `sort()`'s callbacks.